hazelcast-client/src/test/java/com/hazelcast/client/spi/impl/discovery/HazelcastCloudProviderTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.spi.impl.discovery;
+
+import com.hazelcast.logging.LoggingService;
+import com.hazelcast.nio.Address;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.net.UnknownHostException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.hazelcast.logging.Logger.getLogger;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class HazelcastCloudProviderTest {
+
+    private static final String PRIVATE_IP_ADDRESS = "127.0.0.1";
+    private static final String PUBLIC_IP_ADDRESS = "198.51.100.1";
+
+    private HazelcastCloudDiscovery hazelcastCloudDiscovery;
+    private HazelcastCloudAddressProvider provider;
+
+    @Before
+    public void setUp() throws UnknownHostException {
+        Map<Address, Address> addresses = new HashMap<Address, Address>();
+        addresses.put(new Address(PRIVATE_IP_ADDRESS, 5701), new Address(PUBLIC_IP_ADDRESS, 5701));
```

Comment:
can we add minimum of two addresses during the test?

Reply:
OK